spec/rdkafka/consumer_spec.rb

```diff
@@ -271,6 +271,14 @@ def send_one_message(val)
   describe "#close" do
     it "should close a consumer" do
       consumer.subscribe("consume_test_topic")
+      100.times do |i|
+        report = producer.produce(
+          topic:     "consume_test_topic",
+          payload:   "payload #{i}",
+          key:       "key #{i}",
+          partition: 0
+        ).wait
+      end
       consumer.close
```

Comment:
I played around with the tests a bit and realized that this one still passes even if you call close. To get it to fail with the `consumer.close` commented out I had to produce more than `50` messages.  I expected to have to produce at least one message to get it to fail in that state, but needing to produce 100 sort of surprised me, and tells me I need to think carefully about the tests I'm adding, given that they are all about latency and time outs.

Reply:
To make the test fail with consumer.close commented out there is no need to produce new messages, it is enought to raise the poll timeout, e.g.:
expect(consumer.poll(10000)).to be_nil